pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -387,6 +387,18 @@ func (c *Reconciler) resolvePipelineState(
 				return nil, controller.NewPermanentError(err)
 			}
 		}
+
+		if config.FromContextOrDefaults(ctx).FeatureFlags.EnableParamEnum {
+			for _, tr := range resolvedTask.TaskRuns {
+				for _, cond := range tr.Status.Conditions {
+					if cond.Type == apis.ConditionSucceeded && cond.Status == corev1.ConditionFalse && cond.Reason == v1.TaskRunReasonInvalidParamValue {
+						pr.Status.MarkFailed(v1.PipelineRunReasonInvalidParamValue.String(),
+							"Invalid param value in the referenced Task from PipelineTask \"%s\": %s", resolvedTask.PipelineTask.Name, cond.Message)
+						return nil, controller.NewPermanentError(err)
```

Comment:
where does the err in controller.NewPermanentError(err) come from? Looks like it is created in previous code, but not from this code

Reply:
Sorry.. I think I pushed too fast. Probably need to take a rest for the day 😵

Reply:
😄